src/compiler/scala/tools/nsc/ast/parser/Parsers.scala

```diff
@@ -287,8 +275,12 @@ self =>
      */
     @inline final def lookingAhead[T](body: => T): T = {
       val saved = new ScannerData {} copyFrom in
+      val seps  = in.sepRegions
```

Comment:
Do you think this should be part of copyFrom?

Reply:
The scope is clear here: looking ahead into a bracket shouldn't change state. copyFrom is a low-level op used by nextToken, which reasons about sepRegions and mutates next/prev.

The use case here is: inParens, looking ahead sees a close paren, which is legal for empty parens.